pilot/pkg/xds/ads.go

```diff
@@ -817,6 +850,16 @@ func (conn *Connection) send(res *discovery.DiscoveryResponse) error {
 	}
 }
 
+// nolint
+// Synced checks if the type has been synced, meaning the most recent push was ACKed
+func (conn *Connection) Synced(typeUrl string) bool {
+	conn.proxy.RLock()
+	acked := conn.proxy.WatchedResources[typeUrl].NonceAcked
+	sent := conn.proxy.WatchedResources[typeUrl].NonceSent
+	conn.proxy.RUnlock()
+	return acked == sent
```

Comment:
This will not work when envoy NACKed

Reply:
Great point, let me add a test + fix for NACK